dask/array/core.py

```diff
@@ -1919,6 +1973,11 @@ def normalize_chunks(chunks, shape=None):
             raise ValueError(
                 "Chunks and shape must be of the same length/dimension. "
                 "Got chunks=%s, shape=%s" % (chunks, shape))
+    if -1 in chunks:
+        chunks = tuple(s if c == -1 else c for c, s in zip(chunks, shape))
+
+    if any(c == 'auto' for c in chunks):
+        chunks = auto_chunks(chunks, shape, limit, dtype, previous_chunks)
```

Comment:
I feel like at least some of these cases must be mutually exclusive, so `elif`s might be more appropriate to ensure that some corener combination doesn't trigger multiple ones.

Reply:
They generally progress in a way that some of the upper if statements massage data into a form that is appropriate for lower ones.  I don't believe that adding elifs would be helpful here.  However, if you're able to identify a fail case that would be very helpful.